src/coreclr/jit/compiler.cpp

```diff
@@ -4080,6 +4080,15 @@ const char* Compiler::compGetTieringName(bool wantShortName) const
 {
     const bool tier0 = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER0);
     const bool tier1 = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1);
+
+    if (!opts.compMinOptsIsSet)
+    {
+        // If 'compMinOptsIsSet' is not set, just return here. Otherwise, if this method is called
+        // by the assertAbort(), we would recursively call assert while trying to get MinOpts()
+        // and eventually stackoverflow.
+        return opts.compDbgCode ? "Debug" : "Optimization-Level-Not-Yet-Set";
+    }
```

Comment:
Doesn't this need to be under `#ifdef DEBUG`?

Why not remove the `assert(!tier0 || !tier1)` assert, or convert it to:
```
if (tier0 && tier1)
{
  return "Illegal-Both-Tier0-And-Tier1-Set";
}
```
?

Also, why check compDbgCode?

Shouldn't this be just before checking MinOpts(), below?

Reply:
> Unnecessary complexity?

Removed the "Debug" condition.